app/validators/email_address_validator.rb

```diff
@@ -0,0 +1,39 @@
+class EmailAddressValidator < ActiveModel::Validator
+  def validate(record)
+    email_address = record.address
+    unless email_address.nil? || valid_email_address?(email_address)
+      record.errors.add(:address, 'is not an email address')
+    end
+  end
+
+  private
+
+  def valid_email_address?(email_address)
```

Comment:
We should also check it doesn't contain a new line character.

Reply:
Done